src/components/dialogs/ImportData.tsx

```diff
@@ -25,7 +26,7 @@ const ImportDataDialog = ({
   onSubmit,
   columns,
 }: DialogProps & {
-  onSubmit: () => void;
+  onSubmit: (e: any) => void;
```

Comment:
May I know the reason behind this `any`? I assume you're dealing with data with an unknown shape from parsing the CSV? Perhaps consider using `as unknown as ImportedData` or something like that if this is something remotely close to what you're doing.